server/src/test/java/org/elasticsearch/index/mapper/SourceFieldMapperTests.java

```diff
@@ -42,9 +42,25 @@ protected void registerParameters(ParameterChecker checker) throws IOException {
             topMapping(b -> b.startObject(SourceFieldMapper.NAME).field("enabled", false).endObject()),
             dm -> assertFalse(dm.metadataMapper(SourceFieldMapper.class).enabled())
         );
+        checker.registerUpdateCheck(
+            topMapping(b -> b.startObject(SourceFieldMapper.NAME).field("mode", "stored").endObject()),
+            topMapping(b -> b.startObject(SourceFieldMapper.NAME).field("mode", "synthetic").endObject()),
+            dm -> assertTrue(dm.metadataMapper(SourceFieldMapper.class).isSynthetic())
+        );
+        /*
+        checker.registerUpdateCheck(
+            topMapping(b -> b.startObject(SourceFieldMapper.NAME).field("enabled", false).endObject()),
+            topMapping(b -> b.startObject(SourceFieldMapper.NAME).field("mode", "synthetic").endObject()),
+            dm -> assertTrue(dm.metadataMapper(SourceFieldMapper.class).isSynthetic())
+        );
+         */
```

Comment:
this throws exception right? did you mean to test that behaviour?

Reply:
I was trying to see if I could get this update to work, but it isn't possible with the way that merging and validation is implemented at present.  I'll just remove the commented-out lines.